gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiPrimaryOwnerRemovalUpgrader.java

```diff
@@ -0,0 +1,201 @@
+        LOG.warn("");
+        apiIds.forEach(LOG::warn);
+        LOG.warn("");
+        LOG.warn("This can be fixed by editing the apis.primaryOwner.default property of your configuration file");
+        LOG.warn("This value must refer to a valid user or group ID");
+        LOG.warn("");
+        LOG.warn("##############################################################");
+        LOG.warn("");
+    }
+
+    private void warnAndFix(List<String> apiIds, String apiPrimaryOwnerRoleId) throws TechnicalException {
+        warn(apiIds);
+        LOG.info("Attempting to fix APIs without a Primary Owner from configuration");
+        Membership membership = prepareMembership(apiPrimaryOwnerRoleId);
+        for (String apiId : apiIds) {
+            membership.setId(UuidString.generateRandom());
+            membership.setReferenceId(apiId);
+            membershipRepository.create(membership);
+        }
+        String memberType = membership.getMemberType().name().toLowerCase();
+        LOG.info("APIs without a Primary Owner has been associated with {} {}", memberType, defaultPrimaryOwnerId);
+    }
+
+    @Override
+    public int getOrder() {
+        return 100;
+    }
+
+    private String findApiPrimaryOwnerRoleId(String organizationId) throws TechnicalException {
+        return roleRepository
+            .findByScopeAndNameAndReferenceIdAndReferenceType(
+                RoleScope.API,
+                SystemRole.PRIMARY_OWNER.name(),
+                organizationId,
+                RoleReferenceType.ORGANIZATION
+            )
+            .map(Role::getId)
+            .orElseThrow(() -> new TechnicalException("Unable to find API Primary Owner role for organization " + organizationId));
+    }
+
+    private List<String> findEnvironmentIds(String organizationId) throws TechnicalException {
+        return environmentRepository.findByOrganization(organizationId).stream().map(Environment::getId).collect(toList());
+    }
+
+    private List<String> findApiPrimaryOwnerReferenceIds(String apiPrimaryOwnerRoleId, List<String> apiIds) throws TechnicalException {
+        return membershipRepository
+            .findByReferencesAndRoleId(MembershipReferenceType.API, apiIds, apiPrimaryOwnerRoleId)
+            .stream()
+            .map(Membership::getReferenceId)
+            .collect(toList());
+    }
+
+    private Membership prepareMembership(String poRoleId) throws TechnicalException {
+        Optional<User> optUser = userRepository.findById(defaultPrimaryOwnerId);
+        if (optUser.isPresent()) {
+            User user = optUser.get();
+            return membership(user.getId(), MembershipMemberType.USER, poRoleId);
+        }
+
+        return groupRepository
+            .findById(defaultPrimaryOwnerId)
+            .map(group -> membership(group.getId(), MembershipMemberType.GROUP, poRoleId))
+            .orElseThrow(() -> new TechnicalException("Unable to find a user or group with id " + defaultPrimaryOwnerId));
```

Comment:
If this error happens, upgrade() returns false, and I guess it blocks APIM startup.
Do we want that ?
We can imagine that a defaultPrimaryOwner was correctly set in configuration, and the upgrader fixes some anomalies. But a long time after, someone deletes this user and that prevent APIM to start.